controllers/submariner/submariner_controller.go

```diff
@@ -216,7 +216,10 @@ func (r *SubmarinerReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		err := r.client.Status().Update(context.TODO(), instance)
 		if err != nil {
 			reqLogger.Error(err, "failed to update the Submariner status")
-			return reconcile.Result{}, err
+			// Log the error, but indicate success, to avoid reconciliation storms
+			// TODO skitt determine what we should really be doing for concurrent updates to the Submariner CR
+			// Updates fail here because the instance is updated between the .Update() at the start of the function
+			// and the status update here
```

Comment:
Perhaps we should just do one `Update` call and use `CreateOrUpdate` semantics here.

Reply:
The two updates touch two different objects, one inside the other. The first updates the `Submariner` object, before we do much work at all (see the attached comment); the second only updates the status. It seems to me now that if we end up with a version conflict at this point, we _shouldn’t_ update the status at all, because we know that the status is outdated.